socorro/processor/mozilla_transform_rules.py

```diff
@@ -816,3 +816,93 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
         except KeyError:
             return False
         return True
+
+
+#==============================================================================
+class BetaVersionRule(Rule):
+    required_config = Namespace()
+    required_config.add_option(
+        'database_class',
+        doc="the class of the database",
+        default=
+        'socorro.external.postgresql.connection_context.ConnectionContext',
+        from_string_converter=str_to_python_object,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.add_option(
+        'transaction_executor_class',
+        default="socorro.database.transaction_executor."
+                "TransactionExecutorWithInfiniteBackoff",
+        doc='a class that will manage transactions',
+        from_string_converter=str_to_python_object,
+        reference_value_from='resource.postgresql',
+    )
+
+    #--------------------------------------------------------------------------
+    def __init__(self, config):
+        super(BetaVersionRule, self).__init__(config)
+        database = config.database_class(config)
+        self.transaction = config.transaction_executor_class(
+            config,
+            database,
+        )
+
+    #--------------------------------------------------------------------------
+    def version(self):
+        return '1.0'
+
+    #--------------------------------------------------------------------------
+    def _get_versions_data(self):
+        sql = """
+            SELECT
+                pv.product_name,
+                pv.release_version,
+                pv.build_type,
+                pvb.build_id,
+                pv.version_string
+            FROM product_versions pv
+                LEFT JOIN product_version_builds pvb ON
+                    (pv.product_version_id = pvb.product_version_id)
+        """
+        results = self.transaction(
+            execute_query_fetchall,
+            sql
+        )
+        versions_data = {}
+        for product, version, channel, build, real_version in results:
+            key = '%s:%s:%s:%s' % (product, version, channel, build)
+            versions_data[key] = real_version
+
+        return versions_data
+
+    #--------------------------------------------------------------------------
+    def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
+        try:
+            if processed_crash['release_channel'].lower() == 'release':
+                return True
```

Comment:
@rhelmer that's what we do in the middleware, and as far as I remember I was told to do so by Josh. Maybe he told me to do that to be future-ready, in case we started doing rapid aurora and such? 

For context in the middleware, see: https://github.com/mozilla/socorro/blob/master/socorro/external/postgresql/base.py#L400 and https://github.com/mozilla/socorro/blob/master/config/middleware.ini-dist#L472